pkg/validate/validate.go

```diff
@@ -367,16 +367,13 @@ func loadTransform(transformFile string, fs afero.Fs, p *parse.Parser) (*sysl.Ap
 // eg: if grammarFile is ./foo/bar.g, this tries to load ./foo/bar.sysl
 func LoadGrammar(grammarFile string, fs afero.Fs) (*sysl.Application, error) {
 	tokens := strings.Split(grammarFile, string(os.PathSeparator))
-	rootGrammar := strings.Join(tokens[:len(tokens)-1], string(os.PathSeparator))
-	grammarFileName := tokens[len(tokens)-1]
 
-	tokens = strings.Split(grammarFileName, ".")
-	tokens[len(tokens)-1] = "sysl"
-	grammarSyslFile := strings.Join(tokens, ".")
+	grammarFilePieces := strings.Split(tokens[len(tokens)-1], ".")
+	grammarFilePieces[len(grammarFilePieces)-1] = "sysl"
+	tokens[len(tokens)-1] = strings.Join(grammarFilePieces, ".")
 	p := parse.NewParser()
 
-	grammar, name, err := parse.LoadAndGetDefaultApp(grammarSyslFile,
-		syslutil.NewChrootFs(fs, rootGrammar), p)
```

Comment:
I think the purpose of the `NewChrootFs` is so that any imports in the grammar sysl file are resolved relative to the grammar and also restricted to the subtree where the grammar resides. Not using a ChrootFs means the imports in the grammar may now fail because `import /foo` will import from a different path.

We will need to somehow combine the ChrootFs and ModFs. I haven't looked at ChrootFs, but i imagine that because it wraps another FS, it might work anyway, but perhaps there is a strange interaction between that and ModFs.

Reply:
I'm not sure I agree with what I said anymore. If a `.sysl` file is using an absolute import, what should be the root for that import. If you have "import foo" in a file, it should be searched relative to the importing file, but I don't know what "import /foo" should really do. The root is usually defined outside the sysl file. Perhaps it should use the `.sysl` file logic where the presence of a file called `.sysl` defines a root. I don't know.

Reply:
When explicitly add `--root`, `--root` defines a root. Otherwise `.sysl` defines the root by default. btw, `.sysl` is an empty folder according to Novan's implementation.